specs/data_structures.md

```diff
@@ -78,37 +87,36 @@ TODO define a transaction format
 | `blockID`    | [BlockID](#blockid)         |             |
 | `signatures` | [CommitSig](#commitsig)`[]` |             |
 
-## Version
-
-| name                | type     | description |
-| ------------------- | -------- | ----------- |
-| `blockchainVersion` | `uint64` |             |
-| `appVersion`        | `uint64` |             |
-
 ## Time
 
-https://developers.google.com/protocol-buffers/docs/reference/csharp/class/google/protobuf/well-known-types/timestamp
+LazyLedger uses the [Google Protobuf Timestamp](https://developers.google.com/protocol-buffers/docs/reference/csharp/class/google/protobuf/well-known-types/timestamp) format for timestamps, which represents time as seconds in UTC Epoch Time and nanoseconds.
 
 ## BlockID
 
-| name          | type                            | description |
-| ------------- | ------------------------------- | ----------- |
-| `hash`        | [HashDigest](#hashdigest)       |             |
-| `partsHeader` | [PartSetHeader](#partsetheader) |             |
+The block ID is comprised of two distinct Merkle roots:
+1. The root of the [block header](#header)'s fields, in the order provided in the spec.
+1. The root of the complete [serialized](#serialization) block [split into parts](#todo). This is used at the network layer for securely gossiping parts of blocks.
```

Comment:
What's the difference between the partsHeader and the data root?

Reply:
My understanding of the Tendermint design is that it allows blocks to be split into parts that can be securely gossiped in parallel to/from different many different machines. This is an implementation detail really, but apparently an important one. I imagine it might be possible to combine this with the splitting of the erasure coded block in some way (might introduce tight coupling?).

Reply:
Right, this is kind of an unfortunate coupling between the blockchain data structure and the consensus reactor's gossip mechanism - it deserves a new issue and consideration: https://github.com/tendermint/spec/issues/78. Also interested how this relates to erasure coded blocks

Reply:
From my understanding the data root is basically an erasure coded partsHeader, because the leafs of the data root are fixed-sized chunks. This can actually be used to build a more efficient gossiping protocol, using information dispersal algorithms. See section 4.3.1 of https://eprint.iacr.org/2018/460.pdf.

Suppose that you're a validator that just made a block. Instead of gossiping plaintext chunks, you can gossip different erasure coded chunks of the block of your neighbors. The data can still be recovered even if some of your neighbors refuse to gossip your chunks, because you can recover the data from only a subset of the chunks. Therefore you do not need to upload as much data to the network in order to distribute your block.

Reply:
By the way, as mentioned previously, we cannot have multiple Merkle roots pointing to the same data. This is because we cannot guarantee that the data in dataRoot is actually the same as the data in the partsHeader, which would defeat the point of data availability proofs.

Reply:
I've mentioned a very similar idea chatting with @brapse a while ago. I just didn't know about  information dispersal algorithms. It would be awesome if we can implement this as it is needed for LL but could also improve the gossiping mechanism.